models/repo.go

```diff
@@ -1273,6 +1273,17 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			// NOTE: Gogs doesn't actually care about this file so we don't do any error-checking :D
+			os.Remove(daemonExportFile)
+		} else {
+			// NOTE: Gogs doesn't actually care about this file so we don't do any error-checking :D
+			f, _ := os.Create(daemonExportFile)
```

Comment:
I'll add error-logging for `os.Remove` as well, since most admins wanna know if a (now) private repo might still be public... :worried: